mod_wsbridge/mod_wsbridge.c

```diff
@@ -384,6 +450,88 @@ websocket_write_back(struct lws *wsi_in, enum lws_write_protocol type, char *str
 	return n;
 }
 
+void add_content_type(private_t *tech_pvt, cJSON* json) {
+	if (cJSON_GetObjectItem(json, "content-type")) {
+		cJSON_DeleteItemFromObject(json, "content-type");
+	}
+	cJSON_AddItemToObject(json, "content-type", cJSON_CreateString(tech_pvt->content_type));
+}
+
+int is_mute_event(char* event, char* method) {
+	return !strcmp(event, "websocket:media:update") && !strcmp(method, "update");
+}
+
+void reset_ws_write_indexes(private_t *tech_pvt) {
+	switch_mutex_lock(tech_pvt->write_mutex);
+	tech_pvt->write_count = 0;
+	tech_pvt->write_start = 0;
+	tech_pvt->write_end = 0;
+	switch_mutex_unlock(tech_pvt->write_mutex);
+}
+
+char* get_valuestring(cJSON* json, char* field) {
+	char* valuestring = NULL;
+	cJSON* json_field = NULL;
+	json_field = cJSON_GetObjectItem(json, field);
+	if (json_field) {
+		valuestring = json_field->valuestring;
+	}
+	return valuestring;
+}
+
+void on_event(private_t *tech_pvt, cJSON* json) {
+	char* event = NULL;
+	char* method = NULL;
+	event = get_valuestring(json, "event");
+	method =  get_valuestring(json, "method");
+
+	if (!event || !method) {
+		return;
+	}
+
+	if (is_mute_event(event, method)) {
+		cJSON* active;
+		active = cJSON_GetObjectItem(json, "active");
+		if (active && (active->type == cJSON_False || active->type == cJSON_True)) {
+			AudioActive_set(&tech_pvt->audio_active, active->valueint);
+			if(!active->valueint) {
+				reset_ws_write_indexes(tech_pvt);
+			}
+		}
+	}
+}
+
+void send_json_message(struct lws *wsi, cJSON* json_message) {
+	char *message = NULL;
+	if ((message = cJSON_PrintUnformatted(json_message)) != NULL) {
+		char *buf = NULL;
+		asprintf(&buf," %s", message);
+		if (websocket_write_back(wsi, LWS_WRITE_TEXT, buf, strlen(buf)) > 0) {
+			switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_INFO, "WebSockets sending JSON event: %s\n", message);
+		}
+		switch_safe_free(buf);
+	}
+}
+
+void wsbridge_process_message(private_t *tech_pvt, struct lws *wsi)
+{
+	void* pop;
+	if (Queue_pop(&tech_pvt->eventQueue, &pop) == SWITCH_STATUS_SUCCESS) {
+		cJSON *json_message;
+
+		json_message = cJSON_Parse((char *)pop);
+		if (json_message) {
+			// control json event
+			on_event(tech_pvt, json_message);
+			// send json
+			add_content_type(tech_pvt, json_message);
```

Comment:
Also in this case it may not be needed for all events.

Reply:
removed add_content_type